test/e2e/e2e_test.go

```diff
@@ -38,6 +40,128 @@ import (
 
 // +kubebuilder:docs-gen:collapse=Imports
 
+var _ = ginkgo.Describe("Kueue visibility server", func() {
+	const defaultFlavor = "default-flavor"
+	var (
+		defaultRF    *kueue.ResourceFlavor
+		localQueue   *kueue.LocalQueue
+		clusterQueue *kueue.ClusterQueue
+		ns           *corev1.Namespace
+		sampleJob1   *batchv1.Job
+		sampleJob2   *batchv1.Job
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.When("Creating two Jobs with max capacity", func() {
+		ginkgo.BeforeEach(func() {
+			defaultRF = testing.MakeResourceFlavor(defaultFlavor).Obj()
+			gomega.Expect(k8sClient.Create(ctx, defaultRF)).Should(gomega.Succeed())
+
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(defaultFlavor).
+						Resource(corev1.ResourceCPU, "1").
+						Obj(),
+				).
+				Preemption(kueue.ClusterQueuePreemption{
+					WithinClusterQueue: kueue.PreemptionPolicyLowerPriority,
+				}).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+
+			localQueue = testing.MakeLocalQueue("main", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+			gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, defaultRF, true)
+		})
+
+		ginkgo.It("Should fetch information about pending workloads", func() {
+			ginkgo.By("Have zero pending workloads", func() {
+				gomega.Eventually(func() int {
+					info, err := visibilityClient.ClusterQueues().GetPendingWorkloadsSummary(ctx, clusterQueue.Name, metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(info.Items)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(0))
+			})
+
+			ginkgo.By("Schedule two max capacity jobs", func() {
```

Comment:
```suggestion
			ginkgo.By("Schedule a job that when admitted blocks the queue", func() {
```
Maybe split into two steps.  Then, "Schedule a job which is pending" due to lower priority"